src/averageAges.js

```diff
@@ -20,6 +20,12 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  return people
+    .filter(person => (century !== undefined
+      ? Math.ceil(person.died / 100) === century : true) && person.sex === 'm')
+    .map(person => person.died - person.born)
+    .reduce((result, element, index) =>
+      (result * index + element) / (index + 1));
```

Comment:
why do you do extra actions? (like multiplying)

Reply:
to keep the same impact of all elements